packages/shared/src/vis-utils.ts

```diff
@@ -64,24 +64,13 @@ export function formatBool(value: ScalarValue<BooleanType>): string {
 }
 
 export function createComplexFormatter(
-  specifier: string,
-  full = false,
+  formatReal: (val: number) => string,
+  formatImag = formatReal,
 ): (val: ScalarValue<ComplexType>) => string {
-  const formatVal = format(specifier);
-
   return (value) => {
     const [real, imag] = value;
-
-    if (imag === 0 && !full) {
-      return formatVal(real);
-    }
-
-    if (real === 0 && !full) {
-      return `${formatVal(imag)} i`;
-    }
```

Comment:
I'm removing the feature of showing only the real part or only the imaginary part if the other part is equal to 0.

Reply:
Yeah, I don't know why we thought this was a great idea :joy_cat: